src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -92,6 +88,8 @@ BlockIO InterpreterDropQuery::executeToTable(
         if (query_ptr->as<ASTDropQuery &>().is_view && !table->isView())
             throw Exception("Table " + table_id.getNameForLogs() + " is not a View", ErrorCodes::LOGICAL_ERROR);
 
+        table_id = table->getStorageID();
```

Comment:
I think we need to add a comment, why we resolve `table_id` from `query` and one more time here?